File: MCRRTImp/src/main/java/lab/mars/MCRRTImp/model/CircleBound.java
package lab.mars.MCRRTImp.model;

import lab.mars.RRTBase.Obstacle;

public class CircleBound implements Obstacle<Vector2> {

    double radius;

    Vector2 centroid;

    public CircleBound(double x, double y, double radius) {
        this.centroid = new Vector2(x, y);
        this.radius = radius;
    }

    @Override
    public boolean contains(Vector2 o) {
        return o.distance2(centroid) > radius * radius;
    }
}

File: MCRRTImp/src/test/kotlin/infrastructure/Vector2BasedImp/CircleBound.kt
package infrastructure.Vector2BasedImp


import lab.mars.MCRRTImp.base.Obstacle

class CircleBound(x: Double, y: Double, internal var radius: Double) : Obstacle<Vector2> {

    internal var centroid: Vector2 = Vector2(x, y)

    override operator fun contains(o: Vector2): Boolean {
        return o.distance2(centroid) > radius * radius
    }
}
